viadot/sources/sap_rfc.py

```diff
@@ -423,6 +423,16 @@ def _get_alias(self, column: str) -> str:
     def _get_client_side_filter_cols(self):
         return [f[1].split()[0] for f in self.client_side_filters.items()]
 
+    def _exclude_whitespaces(self, df: pd.DataFrame()):
+        """
+        Method to exclude all the spaces in the beginning and ending of a string caused
+        by the functionality of SAP to fill missing characters with whitespaces to reach
+        defined character length.
+        """
+        for column in df.columns:
+            df[column] = df[column].str.strip()
+        return df
+
     def to_df(self):
```

Comment:
Could you please reflect the data stripping in the `to_df()` docstring as well?

Reply:
@TillPickha
[📝 Extended docstring of to_df() method](https://github.com/dyvenia/viadot/pull/627/commits/9d5b63d1fe9ceeefaac1cd579de75ff2a7de4850)

Reply:
1. I feel this should be more precise, as wording such as "to fill missing characters" or "to reach defined character length" is a bit vague. Here's how I would write this:
```python
"""
SAP RFC has a special-case handling of VARCHAR columns, wherein for a column of type VARCHAR(n),
each row of that column will be returned as a string of length n, no matter the length of the actual string. 
If the string is smaller than n, SAP fills in the rest of the string with whitespace. 

For example, if the column is of type VARCHAR(10), and the value is "abc", SAP RFC will return the value "abc       ".

Since this behavior is unexpected for virtually all users, we use this helper function to remove the extra whitespace.
"""
```
So basically I describe the logic using more precise language + provide an example.

2. Please use the same docstring format as elsewhere -- in VSCode, we use the `autodocstring` extension with this setting:
`"autoDocstring.docstringFormat": "google"` that makes this easier.

3. Feels like this could be added to `utils.py` as maybe `strip_df()`?

4. Can you add a test for this?

Reply:
[📝 Changed Docstring of helper function](https://github.com/dyvenia/viadot/pull/627/commits/4c2a425c9ef1cb831a296538e30c0436cb148197)

Reply:
@TillPickha I think you missed the other 3 points

Reply:
This one wasn't addressed yet